src/saveload/details.rs

```diff
@@ -0,0 +1,176 @@
+use std::error::Error;
+
+use serde::de::DeserializeOwned;
+use serde::ser::Serialize;
+
+use {Component, Entity, ReadStorage, SystemData, WriteStorage};
+
+use super::marker::Marker;
+use error::NoError;
+
+#[derive(Serialize, Deserialize)]
+#[serde(bound = "")]
+pub struct EntityData<M: Marker, E, T: Components<M::Identifier, E>> {
```

Comment:
No need for the `Marker` bound

Reply:
Either bound or adding another type parameter for use instead of `M::Identifier`.
And bound them equal in all impls